### PR TITLE
Added soap flakes item and recipe for use in washing machines

### PR DESCRIPTION
**Pull request by yowshi**
#### Summary
SUMMARY: Content "Added soap flaked and a recipe to make them."

#### Purpose of change
It's kind of silly we can't use normal soap in washing machines. People used to use regular soap for such purposes long before detergents were a thing. I tried to do this earlier but did it wrong this is a fixed pull request.

#### Describe the solution
I added soap flakes as an item and a recipe to make them from a bar of soap. 

#### Testing
Couldn't test because the C++ code hadn't been added to allow it

### CODE REVIEW

**Comment by AMurkin** [comment]
You need a valid summary.
SUMMARY: [Category] "[description]"

And format your changes with a [web-linter](http://dev.narc.ro/cataclysm/format.html).

File: data/json/items/chemicals_and_resources.json [hunk 1]
```diff
@@ -1120,6 +1120,20 @@
     "container": "box_small",
     "count": 20
   },
+  {
+  "type": "AMMO",
+  "id": "soap_flakes",
+  "category": "chems",
+  "price": 500,
+  "name": "soap flakes",
```
Comment (Night-Pryanik) [inline data/json/items/chemicals_and_resources.json]: Please add a `name_plural` field.

File: data/json/recipes/recipe_medsandchemicals.json [hunk 2]
```diff
@@ -722,6 +722,22 @@
...
     "flags": [ "ALLOW_ROTTEN" ]
   },
+  {
+  "type": "recipe",
+  "result": "soap_flakes",
+  "category": "CC_CHEM",
+  "subcategory": "CSC_CHEM_CHEMICALS",
+  "skill_used": "cooking",
+  "difficulty": 1,
+  "time": "45 m",
```
Comment (Night-Pryanik) [inline data/json/recipes/recipe_medsandchemicals.json]: 45 minutes to cut flakes?

**Comment by yowshi** [comment]
i swear i ran them through the linter i have no idea why that failed.

File: data/json/recipes/recipe_medsandchemicals.json (hunk 2 above)
Comment (yowshi) [reply data/json/recipes/recipe_medsandchemicals.json]: well you have to cut up a bar of soap into small flakes. I don't think 45 minutes is to long for that.

**Comment by AMurkin** [comment]
You may also add `"looks_like": "detergent"` to the item.

File: data/json/recipes/recipe_medsandchemicals.json (hunk 2 above)
Comment (AMurkin) [reply data/json/recipes/recipe_medsandchemicals.json]: Suggestion: two recipes.
* one like this (I also agree that it should be less than 15 minutes)
* second with a more suitable tool if we have it (several minutes to craft)

File: data/json/recipes/recipe_medsandchemicals.json [hunk 3]
```diff
@@ -722,6 +722,30 @@
...
+    "subcategory": "CSC_CHEM_CHEMICALS",
+    "skill_used": "cooking",
+    "difficulty": 1,
+    "time": "15 m",
+    "autolearn": true,
+    "qualities": [ { "id": "CUT", "level": 1 } ],
+    "components": [ [ [ "soap", 1 ] ] ]
+  },
+  {
+    "type": "recipe",
```
Comment (Rail-Runner) [inline data/json/recipes/recipe_medsandchemicals.json]: This recipe is probably unnecessary; don't think there's any item that provides CUT 2 quality.

File: data/json/recipes/recipe_medsandchemicals.json [hunk 4]
```diff
@@ -722,6 +722,30 @@
     ],
     "flags": [ "ALLOW_ROTTEN" ]
   },
+  {
+    "type": "recipe",
+    "result": "soap_flakes",
+    "category": "CC_CHEM",
+    "subcategory": "CSC_CHEM_CHEMICALS",
+    "skill_used": "cooking",
+    "difficulty": 1,
```
Comment (Rail-Runner) [inline data/json/recipes/recipe_medsandchemicals.json]: I think cutting up a bar of soap is trivial enough to have a difficulty of 0.

Comment (yowshi) [reply data/json/recipes/recipe_medsandchemicals.json]: ok i missed that, my bad.

File: data/json/recipes/recipe_medsandchemicals.json (hunk 3 above)
Comment (yowshi) [reply data/json/recipes/recipe_medsandchemicals.json]: according to amurkin there is one

Comment (AMurkin) [reply data/json/recipes/recipe_medsandchemicals.json]: There are `integrated toolset` and `extended toolset`.
Can they be used for crafting?

File: data/json/recipes/recipe_medsandchemicals.json [hunk 5]
```diff
@@ -722,6 +722,30 @@
     ],
     "flags": [ "ALLOW_ROTTEN" ]
   },
+  {
+    "type": "recipe",
+    "result": "soap_flakes",
+    "category": "CC_CHEM",
+    "subcategory": "CSC_CHEM_CHEMICALS",
+    "skill_used": "cooking",
+    "difficulty": 0,
```
Comment (AMurkin) [inline data/json/recipes/recipe_medsandchemicals.json]:
```suggestion
```

File: data/json/recipes/recipe_medsandchemicals.json [hunk 6]
```diff
@@ -722,6 +722,30 @@
...
+    "qualities": [ { "id": "CUT", "level": 1 } ],
+    "components": [ [ [ "soap", 1 ] ] ]
+  },
+  {
+    "type": "recipe",
+    "result": "soap_flakes",
+    "category": "CC_CHEM",
+    "subcategory": "CSC_CHEM_CHEMICALS",
+    "skill_used": "cooking",
+    "difficulty": 0,
```
Comment (AMurkin) [inline data/json/recipes/recipe_medsandchemicals.json]:
```suggestion
```

File: data/json/recipes/recipe_medsandchemicals.json (hunk 3 above)
Comment (Rail-Runner) [reply data/json/recipes/recipe_medsandchemicals.json]: Integrated toolset comes from having integrated toolset CBM installed; I think it was agreed that it really should never be better than standalone tools (except for "portability part), just like other multitools.

**Comment by anothersimulacrum** [comment]
To test this, you will need to add this new item to this <https://github.com/CleverRaven/Cataclysm-DDA/pull/35670/files#diff-f8fb51ddfefd44b40ceb877f22411dabR153> itemgroup, and attempt to use a washing machine with enough of charges of it.
Once #35670 is merged, of course.

**Comment by anothersimulacrum** [comment]
If the issue were open, this PR would also resolve #29330.

**Comment by yowshi** [comment]
i have no idea how to fix this.

**Comment by yowshi** [comment]
tested with the new JSONized detergent code in place. it works in the washing machine.

**Pull request merged**